examples/trusted-sync/src/main.rs

```diff
@@ -37,10 +40,18 @@ async fn sync(cli_cfg: crate::cli::Cli) -> Result<()> {
     let beacon_url: String =
         cli_cfg.beacon_url.unwrap_or_else(|| std::env::var(BEACON_URL).unwrap());
 
+    // Fetch the Chain ID from the L1 provider.
+    let mut l1_provider = AlloyChainProvider::new_http(l1_rpc_url);
+    let chain_id = l1_provider.chain_id().await.expect("Failed to fetch chain ID from L1 provider");
+    // Get the rollup config for the given chain ID.
+    let available_configs =
+        &[OP_MAINNET_CONFIG, OP_SEPOLIA_CONFIG, BASE_MAINNET_CONFIG, BASE_SEPOLIA_CONFIG];
```

Comment:
Can we upstream a `config_by_chain_id` function in `superchain-registry` instead of this? It's also useful in the client program.